sympy/matrices/matrices.py

```diff
@@ -3912,35 +3975,121 @@ def LUdecomposition_Simple(self, iszerofunc=_iszero):
         LUdecompositionFF
         LUsolve
         """
-        if not self.is_square:
-            raise NonSquareMatrixError(
-                "A Matrix must be square to apply LUdecomposition_Simple().")
-        n = self.rows
-        A = self.as_mutable()
-        p = []
-        # factorization
-        for j in range(n):
-            for i in range(j):
-                for k in range(i):
-                    A[i, j] = A[i, j] - A[i, k] * A[k, j]
-            pivot = -1
-            for i in range(j, n):
-                for k in range(j):
-                    A[i, j] = A[i, j] - A[i, k] * A[k, j]
-                # find the first non-zero pivot, includes any expression
-                if pivot == -1 and not iszerofunc(A[i, j]):
-                    pivot = i
-            if pivot < 0:
-                # this result is based on iszerofunc's analysis of the possible pivots, so even though
-                # the element may not be strictly zero, the supplied iszerofunc's evaluation gave True
-                raise ValueError("No nonzero pivot found; inversion failed.")
-            if pivot != j:  # row must be swapped
-                A.row_swap(pivot, j)
-                p.append([pivot, j])
-            scale = 1 / A[j, j]
-            for i in range(j + 1, n):
-                A[i, j] = A[i, j] * scale
-        return A, p
+
+        if rankcheck:
+            SymPyDeprecationWarning(
+                feature="Keyword argument rankcheck is deprecated. "
+                        "Future LUdecomposition_Simple() will complete "
+                        "regardless rank of input matrix.",
+                issue=9796,
+                deprecated_since_version="1.1").warn()
+
+        if self.rows == 0 or self.cols == 0:
+            # Define LU decomposition of a matrix with no entries as a matrix
+            # of the same dimensions with all zero entries.
+            return self.zeros(self.rows, self.cols), []
+
+        lu = self.as_mutable()
+        row_swaps = []
+
+        pivot_col = 0
+        for pivot_row in range(0, lu.rows-1):
+            # Search for pivot. Prefer entry that iszeropivot determines
+            # is nonzero, over entry that iszeropivot cannot guarantee
+            # is  zero.
+            candidate_pivot_row = None
+            while pivot_col != self.cols:
+                for r in range(pivot_row, self.rows):
+                    iszeropivot = iszerofunc(lu[r, pivot_col])
```

Comment:
Well this is embarrassing, I was calling _find_reasonable_limit in an inner loop, and not the correct outer loop. Those profile timings are meaningless.